### ISSUE TITLE
storageclass failed to bond pvc

### ISSUE THREAD

**Issue by club326** [ceph/ceph-csi#2960]
# Environment
- ceph version:14.2
- Kubernetes cluster version : v1.18.9
- cephcsi:v2.1.0
- csi-node-driver-registrar:v1.3.0
- csi-provisioner:v1.4.0
- csi-snapshotter:v1.2.2
- csi-attacher:v2.1.1
- csi-resizer:v0.5.0

When I tried to create storageclass on ceph, I failed serval times, when I execute

`kubectl logs pod/csi-rbdplugin-provisioner-7c85df56d5-ww5rt -c csi-provisioner`

the logs describe as below

```I0324 07:49:08.727184       1 controller.go:1199] provision "dba-mysql/ceph-satassd-claim" class "kubernetes-csi-rbd-sc": started
I0324 07:49:08.730816       1 event.go:255] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"dba-mysql", Name:"ceph-satassd-claim", UID:"682b25a6-6590-446c-a38e-be68d816bb07", APIVersion:"v1", ResourceVersion:"101502206", FieldPath:""}): type: 'Normal' reason: 'Provisioning' External provisioner is provisioning volume for claim "dba-mysql/ceph-satassd-claim"
W0324 07:49:08.733754       1 topology.go:331] No topology keys found on any node
I0324 07:49:08.733776       1 controller.go:494] CreateVolumeRequest {Name:pvc-682b25a6-6590-446c-a38e-be68d816bb07 CapacityRange:required_bytes:2147483648  VolumeCapabilities:[mount:<fs_type:"ext4" mount_flags:"discard" > access_mode:<mode:SINGLE_NODE_WRITER > ] Parameters:map[clusterID:1ed0bc40-a82d-4085-b5d3-5413995eeba8 csi.storage.k8s.io/controller-expand-secret-name:kubernetes-csi-rbd-secret csi.storage.k8s.io/controller-expand-secret-namespace:ceph-csi csi.storage.k8s.io/fstype:ext4 csi.storage.k8s.io/node-stage-secret-name:kubernetes-csi-rbd-secret csi.storage.k8s.io/node-stage-secret-namespace:ceph-csi csi.storage.k8s.io/provisioner-secret-name:kubernetes-csi-rbd-secret csi.storage.k8s.io/provisioner-secret-namespace:ceph-csi imageFeatures:layering pool:rbd-sata-ssd] Secrets:map[] VolumeContentSource:<nil> AccessibilityRequirements:<nil> XXX_NoUnkeyedLiteral:{} XXX_unrecognized:[] XXX_sizecache:0}
I0324 07:49:08.736449       1 connection.go:180] GRPC call: /csi.v1.Controller/CreateVolume
I0324 07:49:08.736464       1 connection.go:181] GRPC request: {"capacity_range":{"required_bytes":2147483648},"name":"pvc-682b25a6-6590-446c-a38e-be68d816bb07","parameters":{"clusterID":"1ed0bc40-a82d-4085-b5d3-5413995eeba8","imageFeatures":"layering","pool":"rbd-sata-ssd"},"secrets":"***stripped***","volume_capabilities":[{"AccessType":{"Mount":{"fs_type":"ext4","mount_flags":["discard"]}},"access_mode":{"mode":1}}]}
I0324 07:49:08.768145       1 connection.go:183] GRPC response: {}
I0324 07:49:08.768536       1 connection.go:184] GRPC error: rpc error: code = Internal desc = error (an error (exit status 1) occurred while running rados args: [-m 10.1.56.120:6789,10.1.56.121:6789,10.1.56.122:6789 --id rbdsatassd --keyfile=***stripped*** -c /etc/ceph/ceph.conf -p rbd-sata-ssd getomapval csi.volumes.default csi.volume.pvc-682b25a6-6590-446c-a38e-be68d816bb07 /tmp/omap-get-291983009]) occurred, command output streams is ( 2022-03-24T07:49:08.765+0000 7fa383fff700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
2022-03-24T07:49:08.765+0000 7fa38cee0700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
2022-03-24T07:49:08.765+0000 7fa38c6df700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
failed to fetch mon config (--no-mon-config to skip)
)
I0324 07:49:08.768568       1 controller.go:1016] Final error received, removing PVC 682b25a6-6590-446c-a38e-be68d816bb07 from claims in progress
W0324 07:49:08.768580       1 controller.go:887] Retrying syncing claim "682b25a6-6590-446c-a38e-be68d816bb07", failure 8
E0324 07:49:08.768593       1 controller.go:910] error syncing claim "682b25a6-6590-446c-a38e-be68d816bb07": failed to provision volume with StorageClass "kubernetes-csi-rbd-sc": rpc error: code = Internal desc = error (an error (exit status 1) occurred while running rados args: [-m 10.1.56.120:6789,10.1.56.121:6789,10.1.56.122:6789 --id rbdsatassd --keyfile=***stripped*** -c /etc/ceph/ceph.conf -p rbd-sata-ssd getomapval csi.volumes.default csi.volume.pvc-682b25a6-6590-446c-a38e-be68d816bb07 /tmp/omap-get-291983009]) occurred, command output streams is ( 2022-03-24T07:49:08.765+0000 7fa383fff700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
2022-03-24T07:49:08.765+0000 7fa38cee0700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
2022-03-24T07:49:08.765+0000 7fa38c6df700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
failed to fetch mon config (--no-mon-config to skip)
)
I0324 07:49:08.768603       1 event.go:255] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"dba-mysql", Name:"ceph-satassd-claim", UID:"682b25a6-6590-446c-a38e-be68d816bb07", APIVersion:"v1", ResourceVersion:"101502206", FieldPath:""}): type: 'Warning' reason: 'ProvisioningFailed' failed to provision volume with StorageClass "kubernetes-csi-rbd-sc": rpc error: code = Internal desc = error (an error (exit status 1) occurred while running rados args: [-m 10.1.56.120:6789,10.1.56.121:6789,10.1.56.122:6789 --id rbdsatassd --keyfile=***stripped*** -c /etc/ceph/ceph.conf -p rbd-sata-ssd getomapval csi.volumes.default csi.volume.pvc-682b25a6-6590-446c-a38e-be68d816bb07 /tmp/omap-get-291983009]) occurred, command output streams is ( 2022-03-24T07:49:08.765+0000 7fa383fff700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
2022-03-24T07:49:08.765+0000 7fa38cee0700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
2022-03-24T07:49:08.765+0000 7fa38c6df700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
```

when I dig into the pods, log detail like this
```kubectl exec -it csi-rbdplugin-64kbj -n ceph-csi -c csi-rbdplugin bash```
the logs detail 
```kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl kubectl exec [POD] -- [COMMAND] instead.
[root@ctnr /]# rados -m 10.1.56.120:6789,10.1.56.121:6789,10.1.56.122:6789 --id rbdsatassd -c /etc/ceph/ceph.conf --no-mon-config df
couldn't connect to cluster: (1) Operation not permitted
[root@ctnr /]# ls -lrht /etc/ceph
total 8.0K
-rw-r--r-- 1 root root  92 Apr  8  2020 rbdmap
-rw-r--r-- 1 root root   0 Mar 24 08:10 keyring
-rw-r----- 1 root root 182 Mar 24 08:10 ceph.conf
[root@ctnr /]# cd /etc/ceph
[root@ctnr ceph]# ls -lrht
total 8.0K
-rw-r--r-- 1 root root  92 Apr  8  2020 rbdmap
-rw-r--r-- 1 root root   0 Mar 24 08:10 keyring
-rw-r----- 1 root root 182 Mar 24 08:10 ceph.conf
[root@ctnr ceph]# touch ceph.client.rbdsatassd.keyring
[root@ctnr ceph]# ls -lrht
total 8.0K
-rw-r--r-- 1 root root  92 Apr  8  2020 rbdmap
-rw-r--r-- 1 root root   0 Mar 24 08:10 keyring
-rw-r----- 1 root root 182 Mar 24 08:10 ceph.conf
-rw-r--r-- 1 root root   0 Mar 24 08:12 ceph.client.rbdsatassd.keyring
[root@ctnr ceph]# vim ceph.client.rbdsatassd.keyring
bash: vim: command not found
[root@ctnr ceph]# vi ceph.client.rbdsatassd.keyring
[root@ctnr ceph]# ls -lrht
total 12K
-rw-r--r-- 1 root root  92 Apr  8  2020 rbdmap
-rw-r--r-- 1 root root   0 Mar 24 08:10 keyring
-rw-r----- 1 root root 182 Mar 24 08:10 ceph.conf
-rw-r--r-- 1 root root  69 Mar 24 08:13 ceph.client.rbdsatassd.keyring
[root@ctnr ceph]# cat ceph.conf 
[global]
auth_cluster_required = cephx
auth_service_required = cephx
auth_client_required = cephx

# Workaround for http://tracker.ceph.com/issues/23446
fuse_set_user_groups = false
[root@ctnr ceph]# cat /etc/ceph/ceph.conf 
[global]
fsid = 1ed0bc40-a82d-4085-b5d3-5413995eeba8
public_network = 10.1.56.0/24
cluster_network = 10.1.56.0/24
mon_initial_members = ceph-a1-56-120, ceph-a1-56-121, ceph-a1-56-122
mon_host = 10.1.56.120,10.1.56.121,10.1.56.122
auth_cluster_required = cephx
auth_service_required = cephx
auth_client_required = cephx

osd_pool_default_size = 2

osd crush update on start = false
[root@ctnr ceph]# rados -m 10.1.56.120:6789,10.1.56.121:6789,10.1.56.122:6789 --id rbdsatassd -c /etc/ceph/ceph.conf --no-mon-config df
2022-03-24T08:14:21.388+0000 7fc6a9535700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
2022-03-24T08:14:21.388+0000 7fc6a8d34700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
2022-03-24T08:14:21.389+0000 7fc6a3fff700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
couldn't connect to cluster: (13) Permission denied
[root@ctnr ceph]# rados -m 10.1.56.120:6789,10.1.56.121:6789,10.1.56.122:6789 --id rbdsatassd -c /etc/ceph/ceph.conf --no-mon-config df
2022-03-24T08:14:27.134+0000 7f9ba387f700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
2022-03-24T08:14:27.134+0000 7f9ba287d700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
2022-03-24T08:14:27.134+0000 7f9ba307e700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
couldn't connect to cluster: (13) Permission denied
[root@ctnr ceph]# cat  /etc/ceph/ceph.client.rbdsatassd.keyring
[client.rbdsatassd]
    key=ALEThivcQiHxAAnT8vlsg7bjXpLIZs0V5R6Q==
[root@ctnr ceph]# rados -m 10.1.56.120:6789,10.1.56.121:6789,10.1.56.122:6789 --id rbdsatassd -c /etc/ceph/ceph.conf --no-mon-config df
2022-03-24T08:14:47.908+0000 7fd845ade700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
2022-03-24T08:14:47.908+0000 7fd846ae0700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
2022-03-24T08:14:47.908+0000 7fd8462df700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
couldn't connect to cluster: (13) Permission denied
[root@ctnr ceph]# rados -m 10.1.56.120:6789,10.1.56.121:6789,10.1.56.122:6789 --id rbdsatassd --no-mon-config df
2022-03-24T08:15:41.986+0000 7f4146a34700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
2022-03-24T08:15:41.986+0000 7f4147235700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
2022-03-24T08:15:41.986+0000 7f413ffff700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2]
couldn't connect to cluster: (13) Permission denied
```

but when onthe host, there are no problem

```rados -m 10.1.56.120:6789,10.1.56.121:6789,10.1.56.122:6789 --id rbdsatassd --no-mon-config df```
and the logs like this

``` POOL_NAME              USED   OBJECTS CLONES    COPIES MISSING_ON_PRIMARY UNFOUND DEGRADED   RD_OPS      RD   WR_OPS      WR USED COMPR UNDER COMPR 
.rgw.root           512 KiB         4      0         8                  0       0        0      139 139 KiB        8   6 KiB        0 B         0 B 
cephfs_data         234 TiB 351701868      0 703403736                  0       0        0 13226598 965 GiB 35597017  21 TiB        0 B         0 B 
cephfs_metadata     143 GiB   1614047      0   4842141                  0       0        0 25534925  16 TiB 36333781 2.4 TiB        0 B         0 B 
default.rgw.control     0 B         8      0        16                  0       0        0        0     0 B        0     0 B        0 B         0 B 
default.rgw.log         0 B       207      0       414                  0       0        0 52562731  50 GiB 35024370   4 KiB        0 B         0 B 
default.rgw.meta    512 KiB         5      0        10                  0       0        0     1202 1.1 MiB       34  15 KiB        0 B         0 B 
rbd-sata-ssd        154 GiB     25607      0     51214                  0       0        0 16951615 278 GiB 11140117 292 GiB        0 B         0 B 
test_ssd_rbd_pool    10 GiB      1337      0      2674                  0       0        0     4490  10 MiB   608915  37 GiB        0 B         0 B 
tests3.rgw.data     2.3 GiB       790      0      1580                  0       0        0     2387 195 MiB     7515 1.2 GiB        0 B         0 B 
tests3.rgw.index    161 KiB         1      0         2                  0       0        0     5634 7.1 MiB     2806 1.6 MiB        0 B         0 B 

total_objects    353343874
total_used       235 TiB
total_avail      338 TiB
total_space      573 TiB
```

that seems the pod can't read the ceph cluster config from host ,can someone tell me why pod can't exec rados command successful

configmap
```# cat csi-config-map.yaml 
---
apiVersion: v1
kind: ConfigMap
data:
  config.json: |-
    [
      {
        "clusterID": "1ed0bc40-a82d-4085-b5d3-5413995eeba8",
        "monitors": [
          "10.1.56.120:6789",
          "10.1.56.121:6789",
          "10.1.56.122:6789"
        ]
      }
    ]
metadata:
  name: ceph-csi-config
```
secret file

```# cat csi-rbd-secret.yaml 
---
apiVersion: v1
kind: Secret
metadata:
  name: kubernetes-csi-rbd-secret
stringData:
  userID: rbdsatassd
  userKey: ALEThivcQiHxAAnT8vlsg7bjXpLIZs0V5R6Q==
```

**Comment by Rakshith-R** [comment]
@club326 

>cephcsi:v2.1.0

Please update Cephcsi version  
https://github.com/ceph/ceph-csi#known-to-work-co-platforms

https://github.com/ceph/ceph-csi/issues/1995
The fix is in v3.2.2 and v3.3.1+
https://lists.ceph.io/hyperkitty/list/ceph-users@ceph.io/thread/K7LDVS7Y5XQV7ILHC5WUWMXVJ5HX4HU3/

**Comment by club326** [comment]
thank you for your reply, and finally, v3.2.2 is OK!
